### PR TITLE
Convert Labels to be an interface.

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/api/common/Labels.java [hunk 1]
```diff
@@ -34,31 +58,41 @@ public void forEach(BiConsumer<String, String> consumer) {
...
       }
     }
+
+    static Labels sortAndFilterToLabels(Object... data) {
+      return new AutoValue_Labels_ArrayBackedLabels(
+          sortAndFilter(data, /* filterNullValues= */ false));
+    }
+
+    @Override
+    public LabelsBuilder toBuilder() {
```
Comment (trask) [inline api/src/main/java/io/opentelemetry/api/common/Labels.java:68]: same request as https://github.com/open-telemetry/opentelemetry-java/pull/2134/files#r530758547, would you mind making `LabelsBuilder` an interface?

**Comment by jkwatson** [comment]
@marcingrzejszczak FYI this potentially a breaking change

File: api/src/main/java/io/opentelemetry/api/common/Labels.java (hunk 1 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/api/common/Labels.java:68]: done. please take a look

**Review by anuraaga — APPROVED** [review]
Thanks!

**Pull request merged**